src/CommunityToolkit.Maui/Views/Popup/PopupContainer.cs

```diff
@@ -0,0 +1,149 @@
+using System.ComponentModel;
+using CommunityToolkit.Maui.Core;
+using Microsoft.Maui.Controls.PlatformConfiguration;
+using Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;
+
+namespace CommunityToolkit.Maui.Views;
+
+sealed partial class PopupContainer<T> : PopupContainer
+{
+	readonly TaskCompletionSource<PopupResult<T>> taskCompletionSource;
+
+	public PopupContainer(View view, PopupOptions popupOptions, TaskCompletionSource<PopupResult<T>> taskCompletionSource)
+		: this(view as Popup<T> ?? CreatePopupFromView<Popup<T>>(view), popupOptions, taskCompletionSource)
+	{
+	}
+
+	public PopupContainer(Popup<T> popup, PopupOptions popupOptions, TaskCompletionSource<PopupResult<T>> taskCompletionSource)
+		: base(popup, popupOptions, null)
+	{
+		this.taskCompletionSource = taskCompletionSource;
```

Comment:
the `taskCompletionSource` isn't nullable, so I think would be a good idea to add a validation here
```suggestion
		ArgumentNullException.ThrowIfNull(taskCompletionSource);
		this.taskCompletionSource = taskCompletionSource;
```